cpp/src_prims/timeSeries/jones_transform.h

```diff
@@ -0,0 +1,440 @@
+    }
+  }
+}
+
+/**
+* @brief Host Function to batchwise transform the autoregressive coefficients according to "jone's (1980)" transformation 
+*
+* @param params: 2D array where each row represents the AR coefficients of a particular model
+* @param batchSize: the number of models in a batch (number of rows in params)
+* @param pValue: the number of coefficients per model (basically number of columns in params)
+* @param newParams: the transformed params (output)
+* @param allocator: object that takes care of temporary device memory allocation of type std::shared_ptr<MLCommon::deviceAllocator>
+* @param stream: the cudaStream object
+*/
+template <typename DataT, typename IdxT = int>
+void ar_param_transform(const DataT* params, IdxT batchSize, IdxT pValue,
+                        DataT* newParams,
+                        std::shared_ptr<MLCommon::deviceAllocator> allocator,
+                        cudaStream_t stream) {
+  ASSERT(batchSize >= 1 && pValue >= 1, "not defined!");
+
+  IdxT nElements = batchSize * pValue;
+
+  //elementWise transforming the params matrix
+  LinAlg::unaryOp(newParams, params, nElements, PAC<DataT>(), stream);
+
+  //setting the kernel configuration
+  static const int BLOCK_DIM_Y = 1, BLOCK_DIM_X = 256;
+  dim3 numThreadsPerBlock(BLOCK_DIM_X, BLOCK_DIM_Y);
+  dim3 numBlocks(ceildiv<int>(batchSize, numThreadsPerBlock.x), 1);
+
+  //calling the kernel
+  switch (pValue) {
+    case 1:
+      ar_param_transform_kernel<DataT, 1, IdxT, BLOCK_DIM_X, BLOCK_DIM_Y>
+        <<<numBlocks, numThreadsPerBlock, 0, stream>>>(newParams, batchSize);
+      break;
+    case 2:
+      ar_param_transform_kernel<DataT, 2, IdxT, BLOCK_DIM_X, BLOCK_DIM_Y>
+        <<<numBlocks, numThreadsPerBlock, 0, stream>>>(newParams, batchSize);
+      break;
+    case 3:
+      ar_param_transform_kernel<DataT, 3, IdxT, BLOCK_DIM_X, BLOCK_DIM_Y>
+        <<<numBlocks, numThreadsPerBlock, 0, stream>>>(newParams, batchSize);
+      break;
+    case 4:
+      ar_param_transform_kernel<DataT, 4, IdxT, BLOCK_DIM_X, BLOCK_DIM_Y>
+        <<<numBlocks, numThreadsPerBlock, 0, stream>>>(newParams, batchSize);
+      break;
+    default:
+      ASSERT(false, "Unsupported pValue '%d'!", pValue);
+  }
+
+  CUDA_CHECK(cudaPeekAtLastError());
+}
+
+/**
+* @brief Host Function to batchwise transform the autoregressive coefficients according to "jone's (1980)" transformation 
+*
+* @param params: 2D array where each row represents the AR coefficients of a particular model
+* @param batchSize: the number of models in a batch (number of rows in params)
+* @param pValue: the number of coefficients per model (basically number of columns in params)
+* @param newParams: the transformed params (output)
+* @param allocator: object that takes care of temporary device memory allocation of type std::shared_ptr<MLCommon::deviceAllocator>
+* @param stream: the cudaStream object
+*/
+template <typename DataT, typename IdxT = int>
+void ar_param_inverse_transform(

```

Comment:
This looks almost identical to the ar_transform as well as the ma_inv_xform and ma_xform. Is it possible to condense these impls down any further?